ProcessHacker/syssccpu.c

```diff
@@ -851,6 +851,15 @@ PPH_STRING PhSipGetCpuBrandString(
         brandLength = sizeof(brandString) - sizeof(ANSI_NULL);
         brand = PhZeroExtendToUtf16Ex(brandString, brandLength);
     }
+#ifdef _ARM64_
+    else
+    {
+        // TODO: ntoskrnl writes this string in ProcessorNameString registry
+        char cpubrand[sizeof(brandString)] = "Not Available";
+        brandLength = sizeof(brandString) - sizeof(ANSI_NULL);
+        brand = PhZeroExtendToUtf16Ex(cpubrand, brandLength);
+    }
+#else
```

Comment:
This block needs to be reversed. Please use ifndef _ARM64_ and move this below the code for __cpuid.

Ntoskrnl is using _MoveToCoprocessor/_MoveToCoprocessor2 for __cpuid and if the brand isn't located on-chip the cpu brand is also located using ACPI data or SMBIOS data - the GetSystemFirmwareTable function returns the data - If you can confirm what table (ACPI vs SMBIOS) includes the CPU brand then I'll include code to parse the data.

Reply:
OK, I'll edit it. But `_MoveToCoprocessor` (mrc) is for ARM not for ARM64. Also I can not check whether that API returns any value because I use Windows PE and don't have any dedicated Windows ARM PC.

Reply:
> Also I can not check whether that API returns any value because I use Windows PE and don't have any dedicated Windows ARM PC.

again I can test what ever you want. You can also use [QEMU to emulate ARM64, but this will be slow](https://withinrafael.com/2018/02/12/boot-arm64-builds-of-windows-10-in-qemu/) depending on your used CPU.

Reply:
By test, I mean full debugging, breakpoints etc. which I have to do myself. If you can do that in your ARM device PR will be welcomed. BTW, I have Pentium G2020.

Reply:
@MagicAndre1981 

The issue isn't CPU emulation but rather having actual hardware to check the ACPI tables and SMBIOS data provided by the hardware.

Reply:
> @MagicAndre1981
> 
> The issue isn't CPU emulation but rather having actual hardware to check the ACPI tables and SMBIOS data provided by the hardware.

ok, in this case you need to buy real ARM64 device like Surface Pro X, the Lumia is a hack and still work in progress. I don't need any ARM device except my Pixel phone. All other devices I use are x84 AMD/Intel Laptops/PCs and x86 8" Tablet for touch testings of apps.

Reply:
> ok, in this case you need to buy real ARM64 device 

At some point sure - we're under lockdown and everything is closed here in my area for probably the next 6-12 months so we need someone else with hardware who could provide the data.